### PR TITLE
EZP-30390: Removed deprecated ezprice field type

### PR DESCRIPTION
**Pull request by adamwojs**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30390](https://jira.ez.no/browse/EZP-30390)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `master` (`8.0`)
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | yes

Removed `ezprice` field type.

**TODO**:
- [X] Implement feature / fix a bug.
- [X] Implement tests.
- [X] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

**Comment by ViniTou** [comment]
Was it confirmed that ezCommerce is not using that one? :D

**Comment by alongosz** [comment]
> Was it confirmed that ezCommerce is not using that one? :D

Yes. Sorry, we shouldn't discuss this internally probably ;)

**Comment by adamwojs** [comment]
PR updated according to @alongosz suggestions.

File: eZ/Publish/Core/Search/Legacy/Tests/Content/AbstractTestCase.php [hunk 1]
```diff
@@ -102,7 +102,7 @@ protected function getConverterRegistry()
                     'ezdatetime' => new Converter\DateAndTimeConverter(),
                     'ezinteger' => new Converter\IntegerConverter(),
                     'ezstring' => new Converter\TextLineConverter(),
-                    'ezprice' => new Converter\IntegerConverter(),
+                    'ezprice' => new Converter\NullConverter(),
```
Comment (alongosz) [inline eZ/Publish/Core/Search/Legacy/Tests/Content/AbstractTestCase.php]: hmm, why this is still needed?

File: doc/bc/changes-8.0.md [hunk 2]
```diff
@@ -8,3 +8,6 @@ Changes affecting version compatibility with former or future versions.
   while the latest Elasticsearch release is 7.0.
 
   The support for this search engine will be provided once again as a separate bundle.
+  
+* The following field type are not supported any more and have been removed:
```
Comment (alongosz) [inline doc/bc/changes-8.0.md]:
```suggestion
* The following Field Types are not supported any more and have been removed:
```

**Pull request merged**